README.adoc

```diff
@@ -0,0 +1,319 @@
+Now how does one implement a data fetcher?
+
+ifdef::env-github[]
+link:src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[GraphQLVerticle allTasks method]
+endif::env-github[]
+ifndef::env-github[]
+[source,java,role="collapsed"]
+.Java `src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java`
+----
+include::src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[tag=allTasks]
+----
+endif::env-github[]
+
+The `allTasks` data fetcher gets the `uncompletedOnly` parameter from the `DataFetchingEnvironment`.
+Its value is either provided by the client or, as defined in the schema file, set to `true`.
+
+WARNING: Do not block the Vert.x event loop in your data fetchers.
+In this how-to, the data set is small and comes from memory, so it's safe to implement `allTasks` in a blocking fashion.
+When working with databases, caches or web services, make sure your data fetcher returns a `CompletionStage`.
+For further details, please refer to the https://vertx.io/docs/vertx-web-graphql/java/[The Vert.x Web GraphQL Handler documentation].
+
+The code for the `complete` mutation is not much different:
+
+ifdef::env-github[]
+link:src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[GraphQLVerticle complete method]
+endif::env-github[]
+ifndef::env-github[]
+[source,java,role="collapsed"]
+.Java `src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java`
+----
+include::src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[tag=complete]
+----
+endif::env-github[]
+
+It gets the `id` parameter provided by the client, then looks up for the corresponding task.
+If a task is found, it is updated and the mutation returns `true` to indicate success.
+
+Almost there!
+Now let's put things together in the verticle `start` method:
+
+ifdef::env-github[]
+link:src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[GraphQLVerticle start method]
+endif::env-github[]
+ifndef::env-github[]
+[source,java,role="collapsed"]
+.Java `src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java`
+----
+include::src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[tag=start]
+----
+<1> Create Vert.x Web GraphQL handler for the GraphQL-Java object
+<2> Define a Vert.x Web route for GraphQL queries and set the GraphQL handler
+endif::env-github[]
+
+== Running the application
+
+The `GraphQLVerticle` needs a `main` method:
+
+ifdef::env-github[]
+link:src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[GraphQLVerticle main method]
+endif::env-github[]
+ifndef::env-github[]
+[source,java,role="collapsed"]
+.Java `src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java`
+----
+include::src/main/java/io/vertx/howtos/graphql/GraphQLVerticle.java[tag=main]
+----
+<1> Create a `Vertx` context
+<2> Deploy `GraphQLVerticle`
+
+Then you can run the application:
+
+* straight from your IDE or,
+* with Maven: `mvn compile exec:java`, or
+* with Gradle: `./gradlew run` (Linux, macOS) or `gradlew run` (Windows).
+
+=== Listing uncompleted tasks
+
```

Comment:
For all query executions, I'd rather show the interaction with cURL or (better) HTTPie. As it is now we can click and see the result in the browser, but when reading the document the link between the query below and the execution may not be as obvious as if the URLs was in clear ;-)